src/store/inbox/inboxReducer.ts

```diff
@@ -65,22 +12,20 @@ const inboxReducer = (state: InboxRedux = defaultInbox, action): InboxRedux => {
     case TYPES.SET_INBOX:
       return {
         ...state,
-        inbox: action.inbox,
+        chatrooms: action.chatrooms,
       };
-    case TYPES.SET_CHATROOM:
-      return {
-        ...state,
-        chatroom: action.chatroom,
-      };
-    case TYPES.SET_MESSAGES:
+    case TYPES.ADD_MESSAGE:
+      state.chatrooms.forEach(chatroom => {
+        if (chatroom.id === action.chatroomId) {
+          chatroom.chatMessages.push(action.message);
+        }
+      });
       return {
         ...state,
-        messages: action.messages,
```

Comment:
might be more efficient to do this as a map
return {
   ...state,
   chatrooms: state.chatrooms.map(chatroom => {
         if (chatroom.id === action.chatroomId) {
            chatroom.chatMessage.push(action.message);
         }
         return chatroom
   })

https://medium.com/@JeffLombardJr/understanding-foreach-map-filter-and-find-in-javascript-f91da93b9f2c